modules/core/src/main/java/org/apache/synapse/util/InlineExpressionUtil.java

```diff
@@ -197,4 +202,48 @@ private static boolean isValidXML(String stringToValidate) {
         }
         return false;
     }
+
+    /**
+     * Checks whether inline text contains synapse expressions
+     * Inline expressions will be denoted inside ${}
+     * e.g.: ${var.var1}, ${payload.element.id}
+     *
+     * @param inlineText Inline text string
+     * @return true if the string contains inline synapse expressions, false otherwise
+     */
+    public static boolean isInlineSynapseExpressionsContentAware(String inlineText) {
+
+        Matcher matcher = SYNAPSE_EXPRESSION_PLACEHOLDER_PATTERN.matcher(inlineText);
+        while (matcher.find()) {
+            // Extract the expression inside ${...}
+            String placeholder = matcher.group(1);
+            if (placeholder.contains("xpath(") || placeholder.contains("payload.") || placeholder.contains("$.")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Process the inline template and replace the synapse expressions with the resolved values
+     *
+     * @param synCtx   Message Context
+     * @param template Inline template
+     * @return Processed inline template
+     */
+    public static String processInLineSynapseExpressionTemplate(MessageContext synCtx, String template)
+            throws JaxenException {
+
+        Matcher matcher = SYNAPSE_EXPRESSION_PLACEHOLDER_PATTERN.matcher(template);
+        StringBuffer result = new StringBuffer();
+        while (matcher.find()) {
+            // Extract the expression inside ${...}
+            String placeholder = matcher.group(1);
+            SynapseExpression expression = new SynapseExpression(placeholder);
+            String replacement = expression.stringValueOf(synCtx);
```

Comment:
if the stringValueOf fails we get the value "unknown" should we print a warn log outside the while loop saying processed template might have an incorrect result since evaluation of some expressions failed.

Reply:
We cannot distinguish the `unknown` at the mediator level. We may need to print the warn log at the evaluation.